templates/mesh-gateway-deployment.yaml

```diff
@@ -0,0 +1,172 @@
+{{- if .Values.meshGateway.enabled }}
+{{- if not .Values.connectInject.enabled }}{{ fail "connectInject.enabled must be true" }}{{ end -}}
```

Comment:
@adilyse this is what I'm doing to enforce requirements. What do you think?

Reply:
Yeah it's not great but not sure what else to do. I think this is better than not rendering the template since they'll get an error and can fix it rather than wondering why there's no gateways spinning up

Reply:
Agreed!